zebrad/src/commands/start.rs

```diff
@@ -55,13 +55,48 @@ impl StartCmd {
         ));
 
         info!("initializing verifiers");
-        let verifier = zebra_consensus::chain::init(
+        let chain_verifier = zebra_consensus::chain::init(
             config.consensus.clone(),
             config.network.network,
             state.clone(),
         )
         .await;
 
+        let block_verifier =
+            zebra_consensus::block::BlockVerifier::new(config.network.network, state.clone());
+
+        // 1- get the finalized tip
+        let tip = match state.clone().oneshot(zebra_state::Request::Tip).await {
+            Ok(zebra_state::Response::Tip(Some(tip))) => tip,
+            Ok(zebra_state::Response::Tip(None)) => (
+                zebra_chain::block::Height(1),
+                zebra_chain::parameters::genesis_hash(config.network.network),
+            ),
```

Comment:
You can just skip the check if the state is empty, there's no need to try to fake a genesis block.

(Also, the genesis block is at height 0, and an empty state doesn't have any blocks at all. So the substitution will just confuse people who read the code.)

Reply:
Skipping the check will be easier if you put all this new code in an async function.
Then you can just `return Ok(())` here.